streams/common/src/main/java/com/hortonworks/streamline/streams/common/StreamlineEventImpl.java

```diff
@@ -310,15 +326,38 @@ public final void clear() {
         throw new UnsupportedOperationException();
     }
 
+    public static StreamlineEvent fromString(String s) {
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            Map<String, Object> event = mapper.readValue(
+                    s.substring(s.indexOf(TO_STRING_PREFIX) + TO_STRING_PREFIX.length()),
+                    new TypeReference<Map<String, Object>>() {});
+            return StreamlineEventImpl.builder()
```

Comment:
You may already notice, the information for event ID is lost while converting. Do we want to have a way to preserve event ID in StreamlineEventImpl at least internally (only available in class)?

Reply:
In the sampling case, I am not re-building an event, just passing the string part from the logs as is.